crypto/keys/types.proto

```diff
@@ -1,39 +1,40 @@
 syntax = "proto3";
 package cosmos_sdk.crypto.keys.v1;
 
-import "third_party/proto/cosmos-proto/cosmos.proto";
+// import "third_party/proto/cosmos-proto/cosmos.proto";
 import "third_party/proto/gogoproto/gogo.proto";
 
-option go_package                          = "github.com/cosmos/cosmos-sdk/crypto/keys";
+option go_package = "github.com/cosmos/cosmos-sdk/crypto/keys";
+
 option (gogoproto.goproto_getters_all)     = false;
 option (gogoproto.goproto_stringer_all)    = false;
 option (gogoproto.goproto_enum_prefix_all) = false;
 
 // PubKey defines the application-level PubKey type.
 message PubKey {
-  option (gogoproto.goproto_stringer)  = true;
-  option (gogoproto.equal)             = false;
-  option (cosmos_proto.interface_type) = "*github.com/tendermint/tendermint/crypto.PubKey";
+  option (gogoproto.goproto_stringer) = true;
+  option (gogoproto.equal)            = false;
+  // option (cosmos_proto.interface_type) = "*github.com/tendermint/tendermint/crypto.PubKey";
 
   // sum defines a set of all acceptable concrete PubKey implementations.
   oneof sum {
-    PubKeyEd25519           ed25519   = 1;
-    PubKeySecp256k1         secp256k1 = 2;
-    PubKeySr25519           sr25519   = 3;
+    bytes                   ed25519   = 1;
+    bytes                   secp256k1 = 2;
+    bytes                   sr25519   = 3;
     PubKeyMultisigThreshold multisig  = 4;
   }
 }
 
 // PrivKey defines the application-level PrivKey type.
 message PrivKey {
-  option (gogoproto.goproto_stringer)  = true;
-  option (gogoproto.equal)             = false;
-  option (cosmos_proto.interface_type) = "*github.com/tendermint/tendermint/crypto.PrivKey";
+  option (gogoproto.goproto_stringer) = true;
+  option (gogoproto.equal)            = false;
+  // option (cosmos_proto.interface_type) = "*github.com/tendermint/tendermint/crypto.PrivKey";
   // sum defines a set of all acceptable concrete PrivKey implementations.
   oneof sum {
-    PrivKeyEd25519   ed25519   = 1;
-    PrivKeySecp256k1 secp256k1 = 2;
-    PrivKeySr25519   sr25519   = 3;
+    bytes ed25519   = 1;
+    bytes secp256k1 = 2;
+    bytes sr25519   = 3;
   }
 }
```

Comment:
Shouldn't these `oneof`s actually live in `codec/std`? I'd prefer if that's where we keep all `oneof`s for interface types so that zones can overload more easily.

Reply:
@aaronc yes I need to move them. For eg Ethermint defines a `secp256k1` from geth that's different from the one on tendermint.